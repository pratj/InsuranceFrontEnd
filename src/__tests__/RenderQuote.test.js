import { shallow, mount } from "enzyme";
import RenderQuote from "../components/RenderQuote";
import { render, screen } from "@testing-library/react";
import toJson from "enzyme-to-json";

const quoteData = {
  category: "Motor Insurance",
  product: "Car Insurance",
  quoteData: [
    {
      partner: "Tata AIG",
      quote: {
        premium: 400,
      },
    },
  ],
};

const locationData = {
  location: {
    state: {
      quoteData: JSON.stringify(quoteData),
    },
  },
};

test("render renderQuote component", () => {
  shallow(<RenderQuote locationData={locationData} />);
});

test("renders renderQuote component with parent div", () => {
  const wrapper = shallow(<RenderQuote locationData={locationData} />);
  expect(wrapper.find('[data-test="renderQuote"]').length).toBe(1);
});

test("renders `Buy Now` button", () => {
  render(<RenderQuote locationData={locationData} />);
  expect(screen.getByText(/Buy now/i)).toBeInTheDocument();
});

test("accepts renderQuote props", () => {
  const wrapper = mount(<RenderQuote locationData={locationData} />);
  expect(wrapper.props().locationData).toEqual(locationData);
});

test("RenderQuote snapshot", () => {
  const renderQuoteTree = shallow(<RenderQuote locationData={locationData} />);
  expect(toJson(renderQuoteTree)).toMatchSnapshot();
});