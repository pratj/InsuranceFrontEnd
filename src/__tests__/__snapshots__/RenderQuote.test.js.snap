// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RenderQuote snapshot 1`] = `
<div
  data-test="renderQuote"
>
  <WithStyles(ForwardRef(Grid))
    container={true}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      key="0"
      sm={12}
      xs={12}
    >
      <WithStyles(ForwardRef(Card))
        className="makeStyles-root-1"
        key="0"
        style={
          Object {
            "margin": "auto",
            "marginTop": 20,
            "maxWidth": 700,
          }
        }
      >
        <WithStyles(ForwardRef(CardMedia))
          style={
            Object {
              "width": 250,
            }
          }
        />
        <div>
          <WithStyles(ForwardRef(CardContent))>
            <WithStyles(ForwardRef(Typography))
              component="h2"
              gutterBottom={true}
              variant="h5"
            >
              Tata AIG
            </WithStyles(ForwardRef(Typography))>
            <WithStyles(ForwardRef(Divider)) />
            <br />
            <WithStyles(ForwardRef(Typography))
              color="textSecondary"
              component="p"
              variant="body2"
            >
              Premium
              : 
              400
            </WithStyles(ForwardRef(Typography))>
          </WithStyles(ForwardRef(CardContent))>
        </div>
        <ReactStripeCheckout
          ComponentClass="span"
          amount={40000}
          billingAddress={true}
          className="StripeCheckout"
          currency="INR"
          description="Car Insurance By Tata AIG"
          label="Pay With Card"
          locale="auto"
          name="InsuranceBzr"
          reconfigureOnUpdate={false}
          shippingAddress={true}
          token={[Function]}
          triggerEvent="onClick"
        >
          <WithStyles(ForwardRef(Button))
            className="stripebutton"
            onClick={[Function]}
            style={
              Object {
                "alignContent": "center",
                "backgroundColor": "orange",
                "borderRadius": "15px",
                "display": "flex",
                "height": "20%",
                "margin": "auto",
                "marginLeft": "65px",
                "marginTop": "65px",
              }
            }
          >
            Buy Now
          </WithStyles(ForwardRef(Button))>
        </ReactStripeCheckout>
      </WithStyles(ForwardRef(Card))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Snackbar))
    autoHideDuration={6000}
    onClose={[Function]}
    open={false}
  >
    <Alert
      onClose={[Function]}
      severity="error"
    >
      Sorry! Payment has failed
    </Alert>
  </WithStyles(ForwardRef(Snackbar))>
  <div>
    <WithStyles(ForwardRef(Dialog))
      TransitionComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        }
      }
      aria-describedby="alert-dialog-slide-description"
      aria-labelledby="alert-dialog-slide-title"
      keepMounted={true}
      onClose={[Function]}
      open={false}
    >
      <WithStyles(ForwardRef(DialogTitle))
        id="alert-dialog-slide-title"
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        Your payment was Unsuccessful
      </WithStyles(ForwardRef(DialogTitle))>
      <WithStyles(ForwardRef(DialogContent))>
        <p>
          Error!
        </p>
        <WithStyles(ForwardRef(DialogContentText))
          id="alert-dialog-slide-description"
        >
          <p>
            Some error has occured
          </p>
        </WithStyles(ForwardRef(DialogContentText))>
      </WithStyles(ForwardRef(DialogContent))>
      <WithStyles(ForwardRef(DialogActions))>
        <WithStyles(ForwardRef(Button))
          color="primary"
          onClick={[Function]}
        >
          Check out more Insurances
        </WithStyles(ForwardRef(Button))>
        <WithStyles(ForwardRef(Button))
          color="primary"
          onClick={[Function]}
        >
          Close
        </WithStyles(ForwardRef(Button))>
      </WithStyles(ForwardRef(DialogActions))>
    </WithStyles(ForwardRef(Dialog))>
  </div>
</div>
`;
